extensions/message-redaction.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Message redaction
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "James Wheare"
+    email: "james@irccloud.com"
+    period: "2020"
+  -
+    name: "Val Lorentz"
+    email: "progval+ircv3@progval.net"
+    period: "2023"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `message-redaction` capability name.
+Instead, implementations SHOULD use the `draft/message-redaction` capability
+name to be interoperable with other software implementing a compatible
+work-in-progress version.
+
+The final version of the specification will use an unprefixed capability name.
+
+## Introduction
+
+This specification enables messages to be deleted.
+Use cases include retracting accidentally sent messages, moderation,
+and removing a [`+draft/react` client tag][], amongst others.
+These are cosmetic use cases and do not provide any operational security
+guarantees.
+
+## Architecture
+
+### Dependencies
+
+Clients wishing to use these capabilities MUST negotiate the [`message-tags`][]
+capability with the server.
+Additionally, this capability relies on messages being sent with the
+[`msgid`][] tag, and message ids MUST NOT contain spaces.
+Clients SHOULD negotiate the [`echo-message`][] capabilities in order to receive
+message IDs for their own messages, to allow them to be redacted.
+
+
+### Capability
+
+This specification adds the `draft/message-redaction` capability.
+Clients MUST ignore this capability's value, if any.
+
+Implementations that negotiate these capabilities indicate that they are
+capable of handling the respective commands and message tags described below.
+
+
+### Command
+
+To redact a message, a client MUST negotiate the `draft/message-redaction`
+capability and send a `REDACT` command to a target nickname or channel.
+The command is defined as follows:
+
+    REDACT <target> <msgid> [:reason]
+
+Where `<msgid>` is the id of the message to be redacted.
+
+If the client is authorised to delete the message, the server:
+
+* SHOULD forward this `REDACT`, with an appropriate prefix, to the target
+  recipients which negotiated the `draft/message-redaction` capability, in the
+  same way as PRIVMSG messages.
+* MUST not forward this `REDACT` to target recipients which did not negotiated
+  this capability (see "Fallback" below)
+
+### Chat history
+
+After a message is redacted, [`chathistory`][] responses SHOULD either:
+
+* exclude it entirely
+* replace it with the `REDACT` message with the same value as `@msgid` tag
```

Comment:
What does this mean exactly? This seems confusing to handle.

Reply:
In particular: the original author of the message is unknown in this case. If the intent is to communicate that a message was redacted without disclosing the exact original contents, maybe replacing the message body with `[redacted]` would be a better option? (As in, less edge cases to handle for clients.)

```
@msgid=123 :nick!u@h PRIVMSG #channel :[redacted]
@msgid=124 :nick!u@h REDACT #channel 123 :bad example
```

Reply:
```suggestion
* replace its second parameter and/or tags with a placeholder and
  add the `REDACT` message to the response (not counting toward message limits)
```